src/api/cvc4cpp.cpp

```diff
@@ -1280,6 +1280,143 @@ Sort OpTerm::getSort() const
 
 bool OpTerm::isNull() const { return d_expr->isNull(); }
 
+template <>
+std::string OpTerm::getIndices() const
+{
+  CVC4_API_CHECK_NOT_NULL;
+  std::string i;
+  Kind k = intToExtKind(d_expr->getKind());
+
+  if (k == DIVISIBLE_OP)
+  {
+    CVC4::Integer _int = d_expr->getConst<Divisible>().k;
+    i = _int.toString();
+  }
+  else if (k == RECORD_UPDATE_OP)
+  {
+    i = d_expr->getConst<RecordUpdate>().getField();
+  }
+  else
+  {
+    CVC4ApiExceptionStream().ostream() << "Can't get string index from"
```

Comment:
```suggestion
    CVC4_API_CHECK(false) << "Can't get string index from"
```
Your solution is in general fine, but I'd prefer to use a consistent pattern for all checks/throwing API exceptions. It also 'hides' the 'internal' (stream) handling here, which makes it a bit simpler (as a user of the macro you don't have to think about how and why the OStreamVoider is used and when it is fine to not use it).

Reply:
Gotcha, thanks for the feedback!